pkg/ext-proc/main.go

```diff
@@ -77,24 +83,54 @@ func main() {
 	}
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme: scheme,
+		Scheme: scheme.Scheme,
 	})
 	if err != nil {
 		klog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
-	if err := (&backend.LLMServerPoolReconciler{
-		Datastore:      datastore,
-		Scheme:         mgr.GetScheme(),
-		Client:         mgr.GetClient(),
-		ServerPoolName: *serverPoolName,
-		Namespace:      *namespace,
-		Record:         mgr.GetEventRecorderFor("llmserverpool"),
-	}).SetupWithManager(mgr); err != nil {
-		klog.Error(err, "Error setting up LLMServerPoolReconciler")
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		klog.Error(err, "Error building kubernetes clientset")
+		klog.FlushAndExit(klog.ExitFlushTimeout, 1)
 	}
 
+	client, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		klog.Error(err, "Error building kubernetes clientset")
+		klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+	}
+
+	poolInformerFactory := informers.NewSharedInformerFactory(client, time.Second*30)
+
+	poolReconciler := backend.NewLLMServerPoolReconciler(
+		ctx,
+		poolInformerFactory.Api().V1alpha1().LLMServerPools(),
+		client,
+		kubeClient,
+		scheme.Scheme,
+		*serverPoolName,
+		*namespace,
+		*zone,
+		datastore,
+	)
+
+	// This is required, and informers will not sync without this func being called.
+	poolInformerFactory.Start(ctx.Done())
+
+	errChan := make(chan error)
+	go func() {
+		if err := poolReconciler.Run(ctx, 2); err != nil {
```

Comment:
make the number of threads a constant and please mimic controller-runtime defaults